src/text_shaper/directwrite_shaper.cpp

```diff
@@ -671,42 +522,83 @@ std::optional<rasterized_glyph> directwrite_shaper::rasterize(glyph_key _glyph,
 
     auto const width = output.size.width;
     auto const height = output.size.height;
-    output.bitmap.resize(*height * *width * 3);
-    output.format = bitmap_format::rgb;
 
-    std::vector<uint8_t> tmp;
-    tmp.resize(*height * *width * 3);
+    IDWriteFactory2* factory2;
+    IDWriteColorGlyphRunEnumerator* glyphRunEnumerator;
+    if (SUCCEEDED(d->factory->QueryInterface(IID_PPV_ARGS(&factory2))))
+    {
+        hr = factory2->TranslateColorGlyphRun(
+            0.0f,
+            0.0f,
+            &glyphRun,
+            nullptr,
+            DWRITE_MEASURING_MODE::DWRITE_MEASURING_MODE_NATURAL,
+            nullptr,
+            0,
+            &glyphRunEnumerator);
 
-    hr = glyphAnalysis->CreateAlphaTexture(
-        DWRITE_TEXTURE_CLEARTYPE_3x1,
-        &textureBounds,
-        tmp.data(),
-        tmp.size()
-    );
+        if (hr == DWRITE_E_NOCOLOR)
+        {
+            output.bitmap.resize(*height * *width * 3);
```

Comment:
Does this line generate any compiler warning about signedness? Currently the wrapped primitive types are using the element type `int`, but `vector::resize(..)` accepts `size_t` so it should, shouldn't it?

In that case, maybe like that?

```suggestion
            output.bitmap.resize(height.as<size_t>() * width.as<size_t>() * 3);
```

Reply:
There's a lot of unsigned warnings actually... I plan to remove them in future PRs and turn more "warning as error" switch for MSVC build.